Competition2018/src/main/java/competition/subsystems/elevator/commands/TrackCurrentUsageCommand.java

```diff
@@ -0,0 +1,61 @@
+package competition.subsystems.elevator.commands;
+
+import java.util.HashMap;
+
+import com.google.inject.Inject;
+
+import edu.wpi.first.wpilibj.Timer;
+import xbot.common.command.BaseCommand;
+import xbot.common.controls.actuators.XCANTalon;
+
+public class TrackCurrentUsageCommand extends BaseCommand {
+
+    XCANTalon talon;
+    double current;
+    double time;
+    double currentAveragingWindow;
+
+    HashMap<Double, Double> currentHistory;
```

Comment:
This shouldn't be a `HashMap`, nor any other type of `Map`. A map's purpose is to let you to refer to a _value_ by a _key_; you are never going to attempt to look up a measurement by the time it was taken. Instead, I'd suggest an `ArrayDeque` of instances of a new `CurrentSample` class. With an `ArrayDeque` you can `add()` items and then `remove()` when you have the total number of samples that you want.

Reply:
Reading through https://docs.oracle.com/javase/7/docs/api/java/util/ArrayDeque.html that looks much better - it even has nice methods for removing from the end, which will simplify things dramatically.

Reply:
did it!